### PR TITLE
feat: WIP Cypress parallelization

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

Just... had to see if this works... but I doubt it will.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
This does seem to spin up three containers, but they all run ALL the tests. There's no load balancing. I'm trying to understand the docs, but it's unclear. It sounds like the "record" feature is a requirement... that may require a "record key"

¯\\\_(ツ)_/¯

**Comment by ktmud** [comment]
@rusackas The CI bash script already contains logics to enable Cypress parallelization: https://github.com/airbnb/incubator-superset/blob/cda764fd4bd07b41bc06f0783b587c50c99f40b8/.github/workflows/bashlib.sh#L169-L172

All you need to do is adding proper record key to env variables and add the containers matrix.

**Comment by rusackas** [comment]
TODO: Obtain a proper record key, and have Apache Infra store the secret so it can be used as an env var

**Comment by rusackas** [comment]
@ktmud Supposedly the secret is there, and this code sure _looks_ like it should work. I'm clearly missing something here... do the tests need to be restructured / broken up in some way, maybe? This basically _looks_ like the examples in the docs, so I'm not sure why it's running the full test suite on all three machines in the matrix.

**Comment by ktmud** [comment]
@rusackas It's already working. See https://github.com/ktmud/incubator-superset/pull/259 

- Run logs on Github Actions: https://github.com/ktmud/incubator-superset/actions/runs/354501254
- Run logs in Cypress dashboard: https://dashboard.cypress.io/projects/ukwxzo/runs/674/specs 

Apparently the secret hasn't beed added to `apache/incubator-superset` yet. When it's added, you should be able to see
`CYPRESS_RECORD_KEY: ***` in the logs:

(screenshot from my fork)

Currently it's showing as empty in this PR:




Also, someone needs to remove `Cypress (chrome)` from GitHub's  branch protection rules (Settings -> Branches):

**Comment by ktmud** [comment]
It would also be nice to [install the Cypress GitHub app](https://docs.cypress.io/guides/dashboard/github-integration.html#Install-the-Cypress-GitHub-app) so you have PR comments like [this](https://docs.cypress.io/guides/dashboard/github-integration.html#Cypress-GitHub-app-installation-process) (someone would need to request it from Apache via the Cypress dashboard).

File: .github/workflows/superset-e2e.yml [hunk 1]
```diff
@@ -74,8 +79,10 @@ jobs:
           cypress-install
     - name: Run Cypress
       uses: apache-superset/cached-dependencies@b90713b
+      with:
+        parallel: true
+        record: true
```
Comment (robdiciuccio) [inline .github/workflows/superset-e2e.yml:84]: `parallel` and `record` settings here are unnecessary, as they're already enabled in the CI bash script.

**Comment by rusackas** [comment]
Closing this one... @robdiciuccio got it workin' elsewhere! 🎉

**Pull request closed without merging**